server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -1058,6 +1060,24 @@ public IndicesQueryCache getIndicesQueryCache() {
         return indicesQueryCache;
     }
 
+    /**
+     * Increment counter for status
+     *
+     * @param status {@link RestStatus}
+     */
+    public void incDocStatusStats(final RestStatus status) {
```

Comment:
In the TransportUpdateAction as well, we can use ```addDocStatusStats``` itself, it shouldnt incur that much overhead. then we don't need this method as well.